aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -387,21 +393,23 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_cuda_template(const Tensor& input_
     save_mean_ = at::empty({0}, input_options);
     save_invstd_ = at::empty({0}, input_options);
   }
-  auto input = reshaped_packed_accessor<scalar_t, 3>(input_);
-  auto output = reshaped_packed_accessor<scalar_t, 3>(output_);
-  auto weight = reshaped_packed_accessor<scalar_t, 1>(weight_);
-  auto bias = reshaped_packed_accessor<scalar_t, 1>(bias_);
-  auto running_mean = reshaped_packed_accessor<scalar_t, 1>(running_mean_);
-  auto running_var = reshaped_packed_accessor<scalar_t, 1>(running_var_);
-  auto save_mean = reshaped_packed_accessor<accscalar_t, 1>(save_mean_);
-  auto save_invstd = reshaped_packed_accessor<accscalar_t, 1>(save_invstd_);
+  auto input = reshaped_packed_accessor<scalar_t, 3>(input_, true);
+  auto output = reshaped_packed_accessor<scalar_t, 3>(output_, true);
+  auto weight = reshaped_packed_accessor<scalar_t, 1>(weight_, true);
+  auto bias = reshaped_packed_accessor<scalar_t, 1>(bias_, true);
+  auto running_mean = reshaped_packed_accessor<scalar_t, 1>(running_mean_, true);
+  auto running_var = reshaped_packed_accessor<scalar_t, 1>(running_var_, true);
+  auto save_mean = reshaped_packed_accessor<accscalar_t, 1>(save_mean_, true);
+  auto save_invstd = reshaped_packed_accessor<accscalar_t, 1>(save_invstd_, true);
   auto stream = at::cuda::getCurrentCUDAStream();
 
-  constexpr int max_blocks_per_input = 60000;
-  int feature_blocks = std::min<int>(input.size(2), max_blocks_per_input);
-  int batch_blocks   = std::min<int>(input.size(0), max_blocks_per_input / feature_blocks);
-  dim3 blocks(batch_blocks, (input.size(1)+127)/128, feature_blocks);
-  dim3 threads(1, 128);
+  constexpr int max_blocks = 60000;
+  int feature_threads = std::min(getNumThreads(input.size(2)), 128);
```

Comment:
Why is there a `min(*, 128)` here but not below on line 418 when you do the stats kernel?

Reply:
I guess to better phrase the question: why does setting them this way benefit perf?